src/Importer/ModuleImporter.ts

```diff
@@ -116,4 +108,78 @@ export class ModuleImporter implements ImporterContract {
       })
       .join('\n');
   }
+
+  protected transformScript(contents: string, resolveDir: string, sourcefile: string): string {
+    if (contents.includes('import ')) {
+      const { outputFiles } = buildSync({
+        stdin: {
+          contents,
+          resolveDir,
+          sourcefile,
+          loader: 'ts',
+        },
+        platform: 'node',
+        format: 'cjs',
+        external: ['apply'],
+        bundle: true,
+        write: false,
+      });
+
+      return outputFiles[0].text;
+    }
+
+    const { code } = transformSync(contents, {
+      loader: 'ts',
+      format: 'cjs',
+    });
+
+    return code;
+  }
+
+  protected createContext(directory: string, filename: string): Record<string, any> {
+    const exports = {};
+    const moduleGlobals = {
+      exports,
+      require,
+      module: {
+        exports,
+        filename,
+        id: filename,
+        path: directory,
+        require: module.require,
+      },
+      __dirname: directory,
+      __filename: filename,
+    };
+
+    const nodeGlobals = {
+      Buffer,
+      clearImmediate,
+      clearInterval,
+      clearTimeout,
+      console,
+      global,
+      process,
+      queueMicrotask,
+      setImmediate,
+      setInterval,
+      setTimeout,
+      // These globals are not available in TypeScript because they live in DOM.d.ts lib
```

Comment:
> Add an ambiant globals.d.ts in the project to define missing globals (even as any)

Did you test this? Wouldn't that be an issue if the global does not actually exist? I don't think we can partially include `dom.d.ts`. 

Also, how are you using these in your presets?

Reply:
> Did you test this? Wouldn't that be an issue if the global does not actually exist?

Adding an ambiant globals.d.ts will work on the TypeScript side but it has no effect on javascript side. So if the global does not exist, an error will be thrown.

All variables I use here are available in node >= 12 so it's pretty safe on these versions. This project has the following engine `"node": ">=12"` declared in its package.json so I considered it safe.

To be safer, I tried a new version that will work on previous node versions. It also avoid adding an ambiant global.d.ts.
Is it OK for you?

> I don't think we can partially include dom.d.ts.

I agree, I think using the above solution is better.

> Also, how are you using these in your presets?

I'm using only `Buffer` for some custom file manipulations or `console` for some tests. But when I tried to import some npm modules, I had many troubles because of missing globals (mostly `process`, `console` and `Buffer`).

I concluded that we should better virtualize node environment to increase compatibility with npm modules.
To achieve this, I added all module and node globals to vm context.